### PR TITLE
[#3065] - fix metadata copy methods

### PR DESCRIPTION
**Pull request by sblack-usu**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Create a web app and copy it.

### CODE REVIEW

**Comment by sblack-usu** [comment]
@zhiyuli  and @hyi.  I added both of you to review this because both of you have worked on the code the bug is present in.  The problem occurs because the add method only expects a string or an int, but the copy provides the actual metadata object.

File: hs_tools_resource/models.py [hunk 1]
```diff
@@ -212,12 +212,14 @@ def get_supported_res_types_str(self):
...
-
         for res_type in supported_res_types:
             # there are two possibilities for res_type_str values:
             # list of string (during normal create or update) or
             # integer (during creating new version of the resource)
-            if isinstance(res_type, int):
+            if isinstance(res_type, SupportedResTypeChoices):
+                meta_instance.supported_res_types.add(res_type)
+
+            elif isinstance(res_type, int):
```
[the remark this answers is not in the document]
Comment (sblack-usu) [reply hs_tools_resource/models.py]: done

File: hs_tools_resource/models.py [hunk 2]
```diff
@@ -387,7 +389,9 @@ def _add_sharing_status(cls, meta_instance, sharing_status_list):
             # there are two possibilities for res_type_str values:
             # list of string (during normal create or update) or
             # integer (during creating new version of the resource)
-            if isinstance(sharing_status, int):
+            if isinstance(sharing_status, SupportedSharingStatusChoices):
+                meta_instance.sharing_status.add(sharing_status)
+            elif isinstance(sharing_status, int):
```
[the remark this answers is not in the document]
Comment (sblack-usu) [reply hs_tools_resource/models.py]: done

**Comment by cuahsimarko** [comment]
fyi I'm reading all PRs and trying to familiarize with codebase

**Comment by sblack-usu** [comment]
@zhiyuli - could you please review this PR?

**Comment by cuahsimarko** [comment]
retest this,please

**Comment by hydrobuild** [comment]
Can one of the admins verify this patch?

**Comment by zhiyuli** [comment]
@sblack-usu 
I dont see any issue with the changes you made with my old codes...

**Comment by sblack-usu** [comment]
thanks

**Pull request merged**